frontend/app/components/progress.tsx

```diff
@@ -0,0 +1,59 @@
+import type React from 'react';
+
+import * as ProgressPrimitive from '@radix-ui/react-progress';
+
+import { useLanguage } from '~/hooks/use-language';
+import { formatPercent } from '~/utils/string-utils';
+import { cn } from '~/utils/tailwind-utils';
+
+const sizes = {
+  sx: 'h-1',
+  sm: 'h-1.5',
+  base: 'h-2.5',
+  lg: 'h-4',
+  xl: 'h-6',
+};
+
+const variants = {
+  blue: 'bg-blue-600',
+  default: 'bg-gray-600',
+  green: 'bg-green-600',
+  red: 'bg-red-500',
+  yellow: 'bg-yellow-400',
+};
+
+const rootBaseClassName = 'relative w-full overflow-hidden rounded-full bg-gray-200';
+const indicatorBaseClassName = 'h-full w-full flex-1 transition-all';
+
+export interface ProgressProps extends React.ComponentPropsWithoutRef<typeof ProgressPrimitive.Root> {
```

Comment:
Not sure if `ComponentPropsWithoutRef` is the correct type to use here since React v19. @sebastien-comeau do you know?

Reply:
> React ComponentPropsWithoutRef Explained
> Definition and Usage
> ComponentPropsWithoutRef is a react function component that can type their props while excluding the ref attribute. This is particularly useful to ensure an element does not receive a ref as part of its props.
> 
> When to Use ComponentPropsWithoutRef
> You should consider using ComponentPropsWithoutRef when creating a component that wraps an HTML element but should not expose a ref to its parent components. This helps maintain a clear component definition and prevents misuse of the ref.

https://www.dhiwise.com/post/understanding-the-role-of-componentpropswithoutref-in-react

Reply:
But in this case, you might want access to the ref (for example, if you want to focus or scroll to it).

Reply:
I guess, why not.

Reply:
So in this case it should just be `ComponentProps`.